src/SelfRoleManager.ts

```diff
@@ -121,6 +123,7 @@ export class SelfRoleManager extends EventEmitter {
   async registerChannel(channelID: Snowflake, options: ChannelOptions) {
     const channel = await this.client.channels.fetch(channelID);
     if (channel) {
+      this.#listenRoleChanges(options.rolesToEmojis);
```

Comment:
While I understand the idea to add an event listener on the `GuildMemberUpdate` event, I am reluctant to the way you do it here.

The problem is not the fact that you add an event listener. It actually is the fact that whenever a channel is unregistered, there is currently no way to remove the added listener. It would be great if you could add something in the `unregisterChannel` method for this too.

As you need to have a reference to the registered listener, you should probably store it within the channel options...

Reply:
Any idea on how you want to tackle this? @vsantele

Reply:
@HunteRoi The reply was not at [the good place](https://github.com/HunteRoi/discord-selfrole/pull/7#issuecomment-1320897430), but I found a way to solve the issue. 

I'm just not sure about the way I had a "private" member in the interface.